crates/sim2h/src/connection_mgr.rs

```diff
@@ -0,0 +1,319 @@
+use crate::*;
+use std::sync::{Arc, Weak};
+
+/// incoming messages from websockets
+#[derive(Debug)]
+pub enum ConMgrEvent {
+    Disconnect(Lib3hUri, Option<Sim2hError>),
+    ReceiveData(Lib3hUri, WsFrame),
+    ConnectionCount(usize),
+}
+
+/// messages for controlling the connection manager
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug)]
+enum ConMgrCommand {
+    Connect(Lib3hUri, TcpWss),
+    SendData(Lib3hUri, WsFrame),
+    Disconnect(Lib3hUri),
+}
```

Comment:
Just remarking that this is pretty much exactly the enum of Walkman events that get recorded and played back, except for using a WireMessage instead of a WsFrame for data. This will make the walkman integration even smoother.